### DISCUSSION TITLE
improve saving storage

### DISCUSSION THREAD

**Discussion by reebok26** [immich-app/immich#8750] [General]
#feature

Adding another user's photos from a shared album to my timeline should not allocate additional disk space.

Currently, each video or photo is saved again in the database. Just save content once and assign it to users. If a user deletes such a video, it will remove the association. Only the last owner will delete the film from the disk.

This is important if someone (like me) uploads multi-gigabyte videos and shares them with family.

**Comment by alextran1502** [comment]
We don't allocate additional storage for shared content, it is just relational in the database. Perhaps there is a misunderstanding here?

**Reply by reebok26** [reply]
It's great news for me. I asked about this on discord some time ago and someone told me that's how it works. Thank you.
